src/validation.cpp

```diff
@@ -1162,6 +1163,12 @@ bool ReadBlockFromDisk(CBlock& block, const FlatFilePos& pos, const Consensus::P
     if (!CheckProofOfWork(block.GetHash(), block.nBits, consensusParams))
         return error("ReadBlockFromDisk: Errors in block header at %s", pos.ToString());
 
+    // Signet only: check block solution, unless this is the genesis block
+    if (consensusParams.signet_blocks && block.GetHash() != consensusParams.hashGenesisBlock && !CheckBlockSolution(block, consensusParams)) {
```

Comment:
Any reason not to move the `block.GetHash() != consensusParams.hashGenesisBlock` check into `CheckBlockSolution`?

Reply:
There was, but not any more -- fixed, thanks!